src/mail.c

```diff
@@ -0,0 +1,739 @@
+#include "global.h"
+#include "malloc.h"
+#include "bg.h"
+#include "scanline_effect.h"
+#include "palette.h"
+#include "gpu_regs.h"
+#include "window.h"
+#include "text_window.h"
+#include "main.h"
+#include "easy_chat.h"
+#include "mail.h"
+#include "task.h"
+#include "menu.h"
+#include "mail_data.h"
+#include "player_pc.h"
+#include "overworld.h"
+#include "help_system.h"
+#include "menu_helpers.h"
+#include "new_menu_helpers.h"
+#include "graphics.h"
+#include "pokemon_icon.h"
+#include "string_util.h"
+#include "strings.h"
+#include "constants/items.h"
+#include "constants/species.h"
+
+enum MailIconParam
+{
+    MAIL_ICON_NONE = 0,
+    MAIL_ICON_BEAD,
+    MAIL_ICON_DREAM
+};
+
+struct MailEcWordLayout
+{
+    u32 numWordsInLine:2;
+    u32 lineXoffset:6;
+    u32 lineHeight:8;
+};
+
+struct MailAttrStruct
+{
+    u8 numRows;
+    u8 nameY;
+    u8 nameX;
+    u8 messageTop;
+    u8 messageLeft;
+    const struct MailEcWordLayout * linesLayout;
+};
+
+struct MailGfxData
+{
+    const void * pal;
+    const void * tiles;
+    const void * map;
```

Comment:
inconsistent pointer space style

Reply:
Imma keep it. `(void|(struct|union) \w+) \* \w+`, but `v?(u|s|bool)(8|16|32|64) \*\w+`